src/PostJob.ts

```diff
@@ -82,7 +104,38 @@ export default class PostJob {
         return;
     }
 
-    process() {
+    async process() {
+        if (!this.subGroup) {
```

Comment:
consider the benefits from moving the whole block from this _if_ into its own method, or function.